addons/hr_contract/models/hr_contract.py

```diff
@@ -16,23 +16,17 @@ class Employee(models.Model):
     medic_exam = fields.Date(string='Medical Examination Date', groups="hr.group_hr_user")
     vehicle = fields.Char(string='Company Vehicle', groups="hr.group_hr_user")
     contract_ids = fields.One2many('hr.contract', 'employee_id', string='Employee Contracts')
-    contract_id = fields.Many2one('hr.contract', compute='_compute_contract_id', string='Current Contract', help='Latest contract of the employee')
+    contract_id = fields.Many2one('hr.contract', string='Current Contract', help='Latest contract of the employee')
```

Comment:
Check to keep a compute method.

Reply:
When the employee has a new (open) contract, his calendar becomes the one from the contract.
And if a CDI is ended to begin another contract, calendar.leaves should be transferred to the new contract's calendar.
It's easier to detect a contract change if it's not computed.